lib/Expression_gen.ml

```diff
@@ -169,17 +167,17 @@ and gen_misc_special_math_app f =
       Some
         (fun ppf es ->
           if List.length es = 2 then pp_binary ppf "std::max(%a, %a)" es
-          else pp_ordinary_fn ppf f es )
+          else pf ppf "%s(@[<hov>%a@])" f (list ~sep:comma pp_expr) es )
   | "min" ->
       Some
         (fun ppf es ->
           if List.length es = 2 then pp_binary ppf "std::min(%a, %a)" es
-          else pp_ordinary_fn ppf f es )
+          else pf ppf "%s(@[<hov>%a@])" f (list ~sep:comma pp_expr) es )
   | "ceil" ->
       Some
         (fun ppf es ->
           if is_scalar (first es) then pp_unary ppf "std::ceil(%a)" es
-          else pp_ordinary_fn ppf f es )
+          else pf ppf "%s(@[<hov>%a@])" f (list ~sep:comma pp_expr) es )
```

Comment:
why prefer the duplication?

Reply:
To answer my own question, seems like the original abstraction wasn't working for some of the use cases and we weren't gaining a ton with it. This seems fine.

Reply:
Yeah, I pulled out the function for exactly that reason. I think the above could still be tidied up a bit though

Reply:
Yeah. Doesn't have to all be perfect all in one go especially at this stage in the project. Don't worry about it this one. Let me take a look at the expression gen stuff for you...